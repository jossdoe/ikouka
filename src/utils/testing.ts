// This is a proxy-file to allow for global wrappers and modifications
// to the functionality of React Testing Library.
// https://testing-library.com/docs/react-testing-library/setup#custom-render
export * from '@testing-library/react'
