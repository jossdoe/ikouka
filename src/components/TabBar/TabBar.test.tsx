import React from 'react'
import { render, screen } from '../../utils/testing'
import userEvent from '@testing-library/user-event'
import { TabBar, TabBarProps } from './index'

const DIRECTIONS: TabBarProps['direction'][] = ['horizontal', 'vertical']
const SIZES: TabBarProps['size'][] = ['sm', 'md', 'lg']

DIRECTIONS.forEach((direction) => {
  SIZES.forEach((size) => {
    test(`TabBar â€“ ${direction}, ${size}`, async () => {
      const setCurrent = jest.fn()

      const props: TabBarProps = {
        tabs: ['One', 'Two', 'Three'],
        current: 'Two',
        setCurrent,
        direction,
        size
      }

      const { asFragment } = render(<TabBar {...props} />)
      expect(asFragment()).toMatchSnapshot()

      expect(screen.getAllByRole('button')).toHaveLength(3)
      expect(screen.getByText(props.tabs[0])).toBeInTheDocument()
      expect(screen.getByText(props.tabs[1])).toBeInTheDocument()
      expect(screen.getByText(props.tabs[2])).toBeInTheDocument()

      expect(setCurrent.mock.calls).toHaveLength(0)
      await userEvent.click(screen.getByText(props.tabs[0]))
      expect(setCurrent.mock.calls).toHaveLength(1)
      expect(setCurrent.mock.calls[0]).toStrictEqual([props.tabs[0]])

      await userEvent.click(screen.getByText(props.tabs[1]))
      expect(setCurrent.mock.calls).toHaveLength(2)
      expect(setCurrent.mock.calls[1]).toStrictEqual([props.tabs[1]])
    })
  })
})
