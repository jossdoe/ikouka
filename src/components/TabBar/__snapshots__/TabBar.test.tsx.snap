// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar – horizontal, lg 1`] = `
<DocumentFragment>
  <div
    class="inline-flex bg-gray-100 text-gray-600 rounded-lg overflow-hidden whitespace-nowrap"
  >
    <button
      class="px-6 py-2.5 text-lg hover:bg-gray-200"
    >
      One
    </button>
    <button
      class="px-6 py-2.5 text-lg bg-blue-100 text-blue-600"
    >
      Two
    </button>
    <button
      class="px-6 py-2.5 text-lg hover:bg-gray-200"
    >
      Three
    </button>
  </div>
</DocumentFragment>
`;

exports[`TabBar – horizontal, md 1`] = `
<DocumentFragment>
  <div
    class="inline-flex bg-gray-100 text-gray-600 rounded-lg overflow-hidden whitespace-nowrap"
  >
    <button
      class="px-4 py-2 text-base hover:bg-gray-200"
    >
      One
    </button>
    <button
      class="px-4 py-2 text-base bg-blue-100 text-blue-600"
    >
      Two
    </button>
    <button
      class="px-4 py-2 text-base hover:bg-gray-200"
    >
      Three
    </button>
  </div>
</DocumentFragment>
`;

exports[`TabBar – horizontal, sm 1`] = `
<DocumentFragment>
  <div
    class="inline-flex bg-gray-100 text-gray-600 rounded-lg overflow-hidden whitespace-nowrap"
  >
    <button
      class="px-3 py-1.5 text-sm hover:bg-gray-200"
    >
      One
    </button>
    <button
      class="px-3 py-1.5 text-sm bg-blue-100 text-blue-600"
    >
      Two
    </button>
    <button
      class="px-3 py-1.5 text-sm hover:bg-gray-200"
    >
      Three
    </button>
  </div>
</DocumentFragment>
`;

exports[`TabBar – vertical, lg 1`] = `
<DocumentFragment>
  <div
    class="inline-flex bg-gray-100 text-gray-600 rounded-lg overflow-hidden whitespace-nowrap flex-col"
  >
    <button
      class="px-6 py-2.5 text-lg hover:bg-gray-200"
    >
      One
    </button>
    <button
      class="px-6 py-2.5 text-lg bg-blue-100 text-blue-600"
    >
      Two
    </button>
    <button
      class="px-6 py-2.5 text-lg hover:bg-gray-200"
    >
      Three
    </button>
  </div>
</DocumentFragment>
`;

exports[`TabBar – vertical, md 1`] = `
<DocumentFragment>
  <div
    class="inline-flex bg-gray-100 text-gray-600 rounded-lg overflow-hidden whitespace-nowrap flex-col"
  >
    <button
      class="px-4 py-2 text-base hover:bg-gray-200"
    >
      One
    </button>
    <button
      class="px-4 py-2 text-base bg-blue-100 text-blue-600"
    >
      Two
    </button>
    <button
      class="px-4 py-2 text-base hover:bg-gray-200"
    >
      Three
    </button>
  </div>
</DocumentFragment>
`;

exports[`TabBar – vertical, sm 1`] = `
<DocumentFragment>
  <div
    class="inline-flex bg-gray-100 text-gray-600 rounded-lg overflow-hidden whitespace-nowrap flex-col"
  >
    <button
      class="px-3 py-1.5 text-sm hover:bg-gray-200"
    >
      One
    </button>
    <button
      class="px-3 py-1.5 text-sm bg-blue-100 text-blue-600"
    >
      Two
    </button>
    <button
      class="px-3 py-1.5 text-sm hover:bg-gray-200"
    >
      Three
    </button>
  </div>
</DocumentFragment>
`;
